Added documentation for Main Menu UI Updates (#407)
smartdevicelink/sdl_app_library_guides: docs/Displaying a User Interface/Main Menu/index.md

To find out more information on how to create `voiceCommands` see the [related documentation](Speech and Audio/Setting Up Voice Commands).
!@

> Head units supporting RPC v7.1+ may support displaying `secondaryText`, `tertiaryText`, and `secondaryArtwork`. This gives the user a richer experience by displaying more data.

Reviewer: ```suggestion
!!! NOTE
Head units supporting RPC v7.1+ may support displaying `secondaryText`, `tertiaryText`, and `secondaryArtwork`. This gives the user a richer experience by displaying more data. Attempting to set this data on head units that doe not support RPC 7.1+ will result in that data not being displayed to the user. 
!!!
```

@![iOS]
##### Objective-C
```objc
// Create the menu cell

> SDLMenuCell *cell = [[SDLMenuCell alloc] initWithTitle:<#NSString#> icon:<#SDLArtwork#> voiceCommands:<#@[NSString]#> secondaryText:<#(nullable NSString *)#> tertiaryText:<#(nullable NSString *)#> secondaryArtwork:<#(nullable SDLArtwork *)#> handler:^(SDLTriggerSource  _Nonnull triggerSource) {

Reviewer: After reading this initializer, I think that it should be changed in the library to:

```objc
- (instancetype)initWithTitle: secondaryText: tertiaryText: icon: secondaryArtwork: voiceCommands: handler:
```

To find out more information on how to create `voiceCommands` see the [related documentation](Speech and Audio/Setting Up Voice Commands).
!@

!!! NOTE

> Head units supporting RPC v7.1+ may support displaying `secondaryText`, `tertiaryText`, and `secondaryArtwork`. This gives the user a richer experience by displaying more data. Attempting to set this data on head units that doe not support RPC 7.1+ will result in that data not being displayed to the user. 

Reviewer: ```suggestion
Head units supporting RPC v7.1+ may support displaying `secondaryText`, `tertiaryText`, and `secondaryArtwork`. This gives the user a richer experience by displaying more data. Attempting to set this data on head units that do not support RPC 7.1+ will result in that data not being displayed to the user.

To determine if the head unit supports displaying these fields, you can check the `SystemCapabilityManager`'s !@@![iOS]`defaultMainWindowCapability.textFields` / `defaultMainWindowCapability.imageFields`!@@![android,javaEE,javaSE] `getDefaultMainWindowCapability().getTextFields()` / `getDefaultMainWindowCapability().getImageFields()`!@@![iOS, android, javaEE, javaSE] properties after successfully connecting to the head unit. Then check those arrays for objects with the related text / image field names.
!@
```

##### Swift
```swift
// Create the menu cell

> let cell = SDLMenuCell(title: <#String#>, secondaryText: <#T##String?#>, tertiaryText: <#T##String?#>, icon: <#SDLArtwork?#>, secondaryArtwork: <#T##SDLArtwork?#>, voiceCommands: <#[String]?#>) { (triggerSource: SDLTriggerSource) in

Reviewer: ```suggestion
let cell = SDLMenuCell(title: <#String#>, secondaryText: <#String?#>, tertiaryText: <#String?#>, icon: <#SDLArtwork?#>, secondaryArtwork: <#SDLArtwork?#>, voiceCommands: <#[String]?#>) { (triggerSource: SDLTriggerSource) in
```

@![android, javaSE, javaEE]
```java

> // TODO: Update cell to the new constructor method that supports secondaryText, tertiaryText, secondaryArtwork

Reviewer: ```suggestion
// TODO: Update cell to the new constructor method?
```

@![android, javaSE, javaEE]
```java

> // TODO: Update cell to the new constructor method?

Reviewer 2: Updated Java Code Snippet:

```java
// Create the menu cell
MenuCell cell = new MenuCell("Cell text", "Secondary Text", "Tertiary Text", null, null, Collections.singletonList("cell text"), new MenuSelectionListener() {
    @Override
    public void onTriggered(TriggerSource trigger) {
        // Menu item was selected, check the `triggerSource` to know if the user used touch or voice to activate it
        // Handle the Cell's Selection
    }
});

sdlManager.getScreenManager().setMenu(Collections.singletonList(cell));
```

##### Objective-C
```objc
// Create the inner menu cell
SDLMenuCell *cell = [[SDLMenuCell alloc] initWithTitle:<#NSString#> icon:<#SDLArtwork#> voiceCommands:<#@[NSString]#> secondaryText:<#(nullable NSString *)#> tertiaryText:<#(nullable NSString *)#> secondaryArtwork:<#(nullable SDLArtwork *)#> handler:^(SDLTriggerSource  _Nonnull triggerSource) {
    // Menu item was selected, check the `triggerSource` to know if the user used touch or voice to activate it
    <#Handle the cell's selection#>
}];

// Create and set the submenu cell

> SDLMenuCell *submenuCell = [[SDLMenuCell alloc] initWithTitle:<#NSString#> icon:<#SDLArtwork?#> submenuLayout:<#SDLMenuLayout#> subCells:@[cell] secondaryText:<#(nullable NSString *)#> tertiaryText:<#(nullable NSString *)#> secondaryArtwork:<#(nullable SDLArtwork *)#>];

Reviewer: I think this initializer should be changed in the library:

```objc
- (instancetype)initWithTitle: secondaryText: tertiaryText: icon: secondaryArtwork: submenuLayout: subCells:
```

self.sdlManager.screenManager.menu = @[submenuCell];
```

##### Swift
```swift
// Create the inner menu cell

> let cell = SDLMenuCell(title: <#String#>, secondaryText: <#T##String?#>, tertiaryText: <#T##String?#>, icon: <#SDLArtwork?#>, secondaryArtwork: <#T##SDLArtwork?#>, voiceCommands: <#[String]?#>) { (triggerSource: SDLTriggerSource) in

Reviewer: ```suggestion
let cell = SDLMenuCell(title: <#String#>, secondaryText: <#String?#>, tertiaryText: <#String?#>, icon: <#SDLArtwork?#>, secondaryArtwork: <#SDLArtwork?#>, voiceCommands: <#[String]?#>) { (triggerSource: SDLTriggerSource) in
```

    // Menu item was selected, check the `triggerSource` to know if the user used touch or voice to activate it
    <#Handle the cell's selection#>
}

> let submenuCell = SDLMenuCell(title: <#String#>, secondaryText: <#T##String?#>, tertiaryText: <#T##String?#>, icon: <#SDLArtwork?#>, secondaryArtwork: <#T##SDLArtwork?#>, submenuLayout: <#SDLMenuLayout#>, subCells: [cell])

Reviewer: ```suggestion
let submenuCell = SDLMenuCell(title: <#String#>, secondaryText: <#String?#>, tertiaryText: <#String?#>, icon: <#SDLArtwork?#>, secondaryArtwork: <#SDLArtwork?#>, submenuLayout: <#SDLMenuLayout#>, subCells: [cell])
```

sdlManager.screenManager.menu = [submenuCell]
```
!@

@![android, javaSE, javaEE]
```java

> // TODO: Update cell to the new constructor method that supports secondaryText, tertiaryText, secondaryArtwork

Reviewer: ```suggestion
// TODO: Update cell to the new constructor method?
```

    <#Handle the cell's selection#>
}];

// Create and set the submenu cell
SDLMenuCell *submenuCell = [[SDLMenuCell alloc] initWithTitle:<#NSString#> secondaryText:<#(nullable NSString *)#> tertiaryText:<#(nullable NSString *)#> icon:<#SDLArtwork?#> secondaryArtwork:<#(nullable SDLArtwork *)#> submenuLayout:<#SDLMenuLayout#> subCells:@[cell]];
self.sdlManager.screenManager.menu = @[submenuCell];
```

##### Swift
```swift
// Create the inner menu cell
let cell = SDLMenuCell(title: <#String#>, secondaryText: <#String?#>, tertiaryText: <#String?#>, icon: <#SDLArtwork?#>, secondaryArtwork: <#SDLArtwork?#>, voiceCommands: <#[String]?#>) { (triggerSource: SDLTriggerSource) in
    // Menu item was selected, check the `triggerSource` to know if the user used touch or voice to activate it
    <#Handle the cell's selection#>
}

let submenuCell = SDLMenuCell(title: <#String#>, secondaryText: <#String?#>, tertiaryText: <#String?#>, icon: <#SDLArtwork?#>, secondaryArtwork: <#SDLArtwork?#>, submenuLayout: <#SDLMenuLayout#>, subCells: [cell])
sdlManager.screenManager.menu = [submenuCell]
```
!@

@![android, javaSE, javaEE]
```java

> // TODO: Update cell to the new constructor method?

Reviewer 2: Updated Java Code Snippet:

```java
// Create the inner menu cell
MenuCell innerCell = new MenuCell("inner menu cell", "secondary text", "tertiary test", null, null,Collections.singletonList("inner menu cell"), new MenuSelectionListener() {
    @Override
    public void onTriggered(TriggerSource trigger) {
        // Menu item was selected, check the `triggerSource` to know if the user used touch or voice to activate it
        // Handle the cell's selection
    }
});

// Create and set the submenu cell
MenuCell cell = new MenuCell("cell", "secondary text", "tertiary text", MenuLayout.LIST, null, null, Collections.singletonList(innerCell));

sdlManager.getScreenManager().setMenu(Collections.singletonList(cell));
```
